components/renku_data_services/project/api.spec.yaml

```diff
@@ -0,0 +1,382 @@
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+    patch:
+      summary: Update specific fields of an existing project
+      parameters:
+        - in: path
+          name: project_id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ProjectPatch"
+      responses:
+        "200":
+          description: The project
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Project"
+        "404":
+          description: The project does not exist
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+    delete:
+      summary: Remove a project
+      parameters:
+        - in: path
+          name: project_id
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: The project was removed or did not exist in the first place
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+  /version:
+    get:
+      summary: Get the version of the service
+      responses:
+        "200":
+          description: The error
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Version"
+
+components:
+  schemas:
+    Project:
+      type: object
+      description: A Renku native project definition and metadata
+      additionalProperties: false
+      properties:
+        id:
+          $ref: "#/components/schemas/Ulid"
+        name:
+          $ref: "#/components/schemas/Name"
+        slug:
+          $ref: "#/components/schemas/Slug"
+        creation_date:
+          $ref: "#/components/schemas/CreationDate"
+        created_by:
+          $ref: "#/components/schemas/User"
+        repositories:
+          $ref: "#/components/schemas/RepositoriesList"
+        visibility:
+          $ref: "#/components/schemas/Visibility"
+        members:
+          $ref: "#/components/schemas/MembersWithRoles"
+      required:
+        - "id"
+        - "name"
+        - "slug"
+        - "created_by"
+        - "visibility"
+      example:
+        id: "TheProjectULID26Characters"
+        name: "Renku R Project"
+        slug: "r-project"
+        created_by:
+          id: "owner-KC-id"
+        visibility: "public"
+    ProjectPost:
+      type: object
+      description: Project metadata to be created in Renku
+      additionalProperties: false
+      properties:
+        name:
+          $ref: "#/components/schemas/Name"
+        slug:
+          $ref: "#/components/schemas/Slug"
+        created_by:
```

Comment:
I would just put the keycloak user id of the user that created the project here, rather than full set of info about the user. Again an anonymous user who lists a public project should not get a full set of information about the project owner.

Reply:
As discussed, we'll keep this as it is since `User` only includes KeyCloak ID. The other thing is that we can send different responses based on user being logged-in or not.